imod/mf6/simulation.py

```diff
@@ -1046,15 +1056,23 @@ def __repr__(self) -> str:
             content = attrs + ["){}"]
         return "\n".join(content)
 
-    def _generate_gwfgwt_exchanges(self):
+    def _generate_gwfgwt_exchanges(self) -> list[GWFGWT]:
         flow_models = self.get_models_of_type("gwf6")
         transport_models = self.get_models_of_type("gwt6")
-
         # exchange for flow and transport
         exchanges = []
-        if len(flow_models) == 1 and len(transport_models) > 0:
-            flow_model_name = list(flow_models.keys())[0]
-            for transport_model_name in transport_models.keys():
-                exchanges.append(GWFGWT(flow_model_name, transport_model_name))
+
+        for flow_model_name in flow_models:
+            tpt_models_of_flow_model = []
+            flow_model = self[flow_model_name]
+            domain = flow_model.domain
+            for tpt_model_name in transport_models:
+                tpt_model = self[tpt_model_name]
+                if tpt_model.domain.equals(domain):
```

Comment:
I think slightly nicer reads here:

```suggestion
                if tpt_model.domain.equals(flow_model.domain):
```

As it clarifies you are testing equality on the transport model domain and flow model domain.

You can remove the line where ``domain`` is declared.